### DISCUSSION TITLE
Initiator hangs if stopping a second time

### DISCUSSION THREAD

**Discussion by tzachshabtay** [quickfixgo/quickfix#620] [Q&A]
Hi, we upgraded quickfix from 0.6.0 to 0.7.0 and seeing that calling `Stop` on the initiator hangs the second time.
Our app has a single initiator which we use for the entire app lifecycle, when we get a goodnight message we stop the initiator and then call `Start` when the new day starts.
So this worked with 0.6.0 but hangs with 0.7.0 at the end of the second day.

We tracked this break to [this PR](https://github.com/quickfixgo/quickfix/pull/531), `session.Stop` is basically ignored on the second time, and we hang here: https://github.com/quickfixgo/quickfix/blob/3bf2b871219bd9921c206adf59585b920c133538/initiator.go#L147

This is because the session is not stopped and stays running.

It's unclear to me whether this is a bug, or if we're not utilizing the initiator correctly and the break is intentional? Are we only supposed to call `Start`/`Stop` once and recreate it each time?

cc @ackleymi @shipa998

**Comment by tzachshabtay** [comment]
@ackleymi really need guidance here, tried recreating the initiator after stopping it but that's giving me a [duplicate session id error](https://github.com/quickfixgo/quickfix/blob/3bf2b871219bd9921c206adf59585b920c133538/registry.go#L101) so it's really not clear what's the proper way to do this with library now? Thanks.

**Comment by rozaydin** [comment]
I am on **v0.9.0** and facing the exact same issue, 
initial start and stop works. But after this state calling start causes **Duplicate SessionID** error, after this error calling stop causes it to hang forever.

**Comment by claytn** [comment]
I'm on the latest version v0.9.4 and also running into this issue. After the initiator has been stopped once and restarted, the second call to Stop hangs forever.

**Reply by claytn** [reply]
It seems like the best course of action here is to create a new initiator instance after each `Stop` call. In one of the more recent changes [here](https://github.com/quickfixgo/quickfix/pull/637), sessions are unregistered after `Stop` is called which means you can recreate a new initiator instance with the same quickfix settings as your first instance (just makes things more convenient).
